File: python/Arrays/day4/zero.py
n = int(input("Enter the size of array: "))

arr = list(map(int, input(f"Enter {n} numbers: ").split()))

# 2 pointer...
j = -1
for i in range(n):
    if arr[i] == 0:
        j = i
        break

for i in range(j+1, n):
    if arr[i] != 0:
        arr[i], arr[j] = arr[j], arr[i]
        j += 1

print("Array after moving zeros to the end: ", arr)

File: java/Arrays/Day4/Zero.java
package Day4;
import java.util.*;

public class Zero {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter the size of the array: ");
        int n = sc.nextInt();

        int arr[] = new int[n];
        System.out.print("Enter " + n + " " + "numbers: ");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }


        // 2 pointer approach for zero...
        int j = -1;
        for (int i = 0; i < n; i++) {
            if (arr[i] == 0) {
                j = i;
                break;
            }
        }

        for (int i = j + 1; i < n; i++) {
            if (arr[i] != 0) {
                swap(arr, i, j);
                j++;
            }
        }

        System.out.println("Array after moving zeros to the end:");
        System.out.println(Arrays.toString(arr));

        sc.close();

    }


    static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
}